listeners/write_file.js

```diff
@@ -0,0 +1,63 @@
+/* 
+ * write_file.js, WRITE FILE Listener - for WRITE FILE
+ * 
+ * (C) 2024 TekMonks. All rights reserved.
+ */
+
+const urlMod = require("url");
+const fs = require('fs');
+const httpServerFactory = require(`${CONSTANTS.LIBDIR}/httpServerFactory.js`);
+
+exports.start = (routeName, listener, messageContainer) => {
+    if ((listener.flow.env[routeName] && listener.flow.env[routeName].server) ||
+        (listener.flow.env[routeName] && listener.flow.env[routeName].creatingServer)) return; // already listening or creating
+
+    listener.flow.env[routeName] = { "creatingServer": true };
+
+
+    httpServerFactory.createHTTPServer(listener, (err, server) => {
+        if (err) {
+            LOG.error(`[REST_LISTENER] Unable to create server due to ${err}, disabling the flow`);
+            delete listener.flow.env[routeName].creatingServer; throw (err);
+        }
+        listener.flow.env[routeName] = { server };
+        delete listener.flow.env[routeName].creatingServer; // done
+
+        listener.flow.env[routeName].server.on("request", (req, res) => {
+            let endPoint = urlMod.parse(req.url, true).pathname;
+            if (endPoint != listener.url) return;   // not ours to handle
+
+            let chunks = [];
+
+            req.on('data', chunk => {
+                chunks.push(chunk);
+            });
+            let buffer;
+            req.on('end', () => {
+                try {
+                    buffer = Buffer.concat(chunks);
```

Comment:
This is also bad code. Bad request would be chunk being null which should be an IF condition not an exception catch and certainly not here. This is just wrong!